Panic in `TestIntegrations`

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/4628180314/jobs/8186957687


#### Relevant snippet

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1f5991a]

goroutine 42584 [running]:
github.com/gravitational/teleport/lib/sshutils.(*Server).HandleConnection(0xc0022aae00, {0xb048388?, 0xc003b6ad20})
	/__w/teleport/teleport/lib/sshutils/server.go:569 +0x177a
created by github.com/gravitational/teleport/lib/sshutils.(*Server).acceptConnections
	/__w/teleport/teleport/lib/sshutils/server.go:460 +0x56a
FAIL	github.com/gravitational/teleport/integration	85.327s
```

--- a/lib/sshutils/server.go
+++ b/lib/sshutils/server.go
@@ -565,6 +565,10 @@ func (s *Server) HandleConnection(conn net.Conn) {
 			for {
 				select {
 				case req := <-reqs:
+					if req == nil {
+						connClosed()
+						return
+					}
 					// wait for a request that wants a reply to send the error
 					if !req.WantReply {
 						continue

--- a/lib/sshutils/server.go
+++ b/lib/sshutils/server.go
@@ -567,7 +567,7 @@ func (s *Server) HandleConnection(conn net.Conn) {
 				case req := <-reqs:
 					if req == nil {
 						connClosed()
-						return
+						break
 					}
 					// wait for a request that wants a reply to send the error
 					if !req.WantReply {

